tests/backup/backup_upgrade_test.go

```diff
@@ -759,3 +760,486 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", func() {
+			schedulePolicyName = fmt.Sprintf("interval-%v-%v", intervalInMins, time.Now().Unix())
+			schedulePolicyInfo := Inst().Backup.CreateIntervalSchedulePolicy(5, int64(intervalInMins), 5)
+			err = Inst().Backup.BackupSchedulePolicy(schedulePolicyName, uuid.New(), BackupOrgID, schedulePolicyInfo)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule policy [%s] with interval [%v] mins", schedulePolicyName, intervalInMins))
+			schedulePolicyUid, err = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, schedulePolicyName)
+			log.FailOnError(err, "Fetching uid of schedule policy [%s]", schedulePolicyName)
+			log.Infof("Schedule policy [%s] uid: [%s]", schedulePolicyName, schedulePolicyUid)
+		})
+		Step("Create schedule backup for each namespace", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			createSingleNSBackupTask := func(appName string) {
+				namespace := srcClusterAppNamespaces[appName][0]
+				labelSelectors := make(map[string]string, 0)
+				singleNSScheduleName := fmt.Sprintf("%s-single-namespace-schedule-%v", namespace, time.Now().Unix())
+				log.InfoD("Creating schedule backup with schedule [%s] of source cluster namespace [%s]", singleNSScheduleName, namespace)
+				err = CreateScheduleBackup(singleNSScheduleName, SourceClusterName, backupLocationName, backupLocationUid, []string{namespace},
+					labelSelectors, BackupOrgID, preRuleNames[appName], preRuleUids[appName], postRuleNames[appName], postRuleUids[appName], schedulePolicyName, schedulePolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup with schedule [%s]", singleNSScheduleName))
+				firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, singleNSScheduleName, BackupOrgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching name of the first schedule backup with schedule [%s]", singleNSScheduleName))
+				log.Infof("First schedule backup name: [%s]", firstScheduleBackupName)
+				singleNSNamespaces = SafeAppend(&mutex, singleNSNamespaces, namespace).([]string)
+				singleNSScheduleNames = SafeAppend(&mutex, singleNSScheduleNames, singleNSScheduleName).([]string)
+			}
+			_ = TaskHandler(Inst().AppList, createSingleNSBackupTask, Parallel)
+		})
+		var versions []string
+		if len(Inst().SchedUpgradeHops) > 0 {
+			versions = strings.Split(Inst().SchedUpgradeHops, ",")
+		}
+		dash.VerifyFatal(len(versions) > 0, true, "Check if upgrade versions provided is provided")
+
+		for _, version := range versions {
+			Step("Upgrading OCP cluster", func() {
+
+				err := SwitchBothKubeConfigANDContext("source")
+				dash.VerifyFatal(err, nil, "Switching context and kubeconfig to source cluster")
+
+				err = Inst().S.UpgradeScheduler(version)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("verify [%s] upgrade to [%s] is successful", Inst().S.String(), version))
+				PrintK8sCluterInfo()
+
```

Comment:
after upgrading dont you need to validate if backup is running as expected?

Reply:
We are doing the ops later on in the testcase which covers it inherently checking it here will always fails as we need to wait for px to first stabalise.